ui/analyse/src/ground.ts

```diff
@@ -6,12 +6,27 @@ import { DrawShape } from 'chessground/draw';
 import resizeHandle from 'common/resize';
 import AnalyseCtrl from './ctrl';
 import * as Prefs from 'common/prefs';
+import { ctrl as makeKeyboardMove } from 'keyboardMove';
 
 export const render = (ctrl: AnalyseCtrl): VNode =>
   h('div.cg-wrap.cgv' + ctrl.cgVersion.js, {
     hook: {
       insert: vnode => {
         ctrl.chessground = site.makeChessground(vnode.elm as HTMLElement, makeConfig(ctrl));
+        if (ctrl.data.pref.keyboardMove) {
+          ctrl.keyboardMove ??= makeKeyboardMove({
+            ...ctrl,
+            data: { ...ctrl.data, player: { color: 'both' } },
+            getCrazyhousePockets: () => ctrl.node.crazy?.pockets,
+            flipNow: ctrl.flip,
+          });
+          ctrl.keyboardMove.update({
+            fen: ctrl.node.fen,
+            canMove: true,
+            cg: ctrl.chessground,
+          });
```

Comment:
ctrl sub-ctrls should be instanciated within the ctrl, not in a view.

Reply:
That definitely feels better. I ended up mimicking how we do things in ui/round: https://github.com/lichess-org/lila/pull/15019/commits/fcbf24f59b6619fe64e81584e1fdda71efb82f40